staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -63,13 +66,27 @@ const (
 	dbMetricsMonitorJitter = 0.5
 )
 
+// TODO(negz): Stop using a package scoped logger. At the time of writing we're
+// creating an etcd client for each CRD. We need to pass each etcd client a
+// logger or each client will create its own, which comes with a significant
+// memory cost (around 20% of the API server's memory when hundreds of CRDs are
+// present). The correct fix here is to not create a client per CRD. See
+// https://github.com/kubernetes/kubernetes/issues/111476 for more.
+var etcd3ClientLogger *zap.Logger
+
 func init() {
 	// grpcprom auto-registers (via an init function) their client metrics, since we are opting out of
 	// using the global prometheus registry and using our own wrapped global registry,
 	// we need to explicitly register these metrics to our global registry here.
 	// For reference: https://github.com/kubernetes/kubernetes/pull/81387
 	legacyregistry.RawMustRegister(grpcprom.DefaultClientMetrics)
 	dbMetricsMonitors = make(map[string]struct{})
+
+	l, err := logutil.CreateDefaultZapLogger(zapcore.InfoLevel)
+	if err != nil {
+		l = zap.NewNop()
+	}
+	etcd3ClientLogger = l
```

Comment:
generally this looks fine as a stop-the-bleeding solution. Only open question as discussed on Slack: somewhere is an integration of this logger into klog if I remember right. Would double check that this change does not influence that. Probably it won't.

Reply:
cc @wojtek-t

Reply:
FWIW this is where the logger was previously being setup - https://github.com/etcd-io/etcd/blob/v3.5.4/client/v3/client.go#L374

Reply:
...which makes me think I should probably give the logger a name as they do in the original code and potentially honor the debug level env var too.

Reply:
Done. I did push the `ETCD_CLIENT_DEBUG` logic in a separate commit just because it looks strange to me - unless I'm missing something it returns the info (default) debugging level no matter what the env var is set to.